### ISSUE TITLE
Why not use Sigmoid over the logit produced by the Classifier in train_matrix_factorization.py?

### ISSUE THREAD

**Issue by thethinkmachine** [lm-sys/RouteLLM#57]
From what I understand, Sigmoid could have been used on top of the classifier to output some value in [0, 1], which could be then compared to the cost treshold/alpha to determine a suitable model.

Also, why not use Softmax when num_classes > 1?
